### PR TITLE
✨ API: new `DELETE /solvers/*/versions/*/jobs/*`

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ✨ implements ``DELETE /solvers/*/versions/*/jobs/*`` entrypoint
- ♻️ Adapted webserver plugin:
  - new client wrapper
  - error handling
- ♻️ Improves OAS error response: adds an error model
- 🔨 Improves tests: Uses the new capture to create backend response mocks (see ``tests/mocks`` folder) in given workflows. These captures are parametrized in some tests to adapt to changing values e.g. `project_id`.

NOTE: this feature is still not exposed in the OAS and works only under `APISERVER_DEV_FEATURE_ENABLED=1` (which is by default in `master` deploy)

## Related issue/s

- resolves https://github.com/ITISFoundation/osparc-simcore/issues/4111
- resolves https://github.com/ITISFoundation/osparc-issues/issues/733

## How to test
```cmd
cd services/api-server
make install-dev
pytest -vv --pdb tests/unit/api_solvers/test_api_routers_solvers_jobs.py::test_delete_solver_job
```

### CODE REVIEW

**Review by sanderegg — COMMENTED** [review]
nice!

**Review by sanderegg — APPROVED** [review]

**Review by bisgaard-itis — APPROVED** [review]
Looks good to me. Just a few comments.

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/api-server/tests/mocks/delete_project_not_found.json
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/tests/mocks/delete_project_not_found.json:1]: Not for the moment but that is also the plan.

File: services/api-server/tests/unit/api_solvers/test_api_routers_solvers_jobs_delete.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+
+class MockedBackendApiDict(TypedDict):
+    catalog: MockRouter | None
+    webserver: MockRouter | None
+
+
+@pytest.fixture
+def solver_key() -> str:
+    return "simcore/services/comp/itis/sleeper"
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/tests/unit/api_solvers/test_api_routers_solvers_jobs_delete.py:27]: yes

**Review by GitHK — APPROVED** [review]
I would have preferred to have tests in place for the models. Hope they will be coming in the future.

**Pull request merged**